config.js

```diff
@@ -247,6 +249,30 @@ function setRankedUnrankedOptionsDefaults(rankedUnrankedOptions, argv) {
     }
 }
 
+function getValidFilename(filename) {
+    // convert any other character than letters (A-Z a-z), numbers (0-9), hypens (-), underscore (_)
+    // to a hyphen (-)
```

Comment:
Actually regex \w means

> Matches any word character (alphanumeric & underscore). Only matches low-ascii characters (no accented or non-roman characters). Equivalent to [A-Za-z0-9_]

To also match hypen you would need `[^\w\-]`. (however since we convert to hypen this wont change the output.
Also we might want to allow dot and space, adding up to `[^\w\-. ]`

Reply:
ah i see!!! i definitely want to learn regexp now 😮 

i think allowing space would be a bad idea (to begin with how do we add a space in the command line args), but the dots why not, even though just for a text filename that may be uneeded to have dots in name

Reply:
dot and space re valid filename charchers. What's bad about them?

Reply:
i thought space would be harder to use correctly in shell command line, and dot may change file extension in an unwanted way, but if these are not issues i can add space and dot too

Reply:
fixed